src/main/scala/io/qbeast/spark/delta/IndexStatusBuilder.scala

```diff
@@ -91,13 +95,34 @@ private[delta] class IndexStatusBuilder(
       .select(
         createCube(col("cube"), lit(ndims)).as("cubeId"),
         col("maxWeight"),
-        normalizeWeight(col("maxWeight"), col("elementCount"), lit(rev.desiredCubeSize)).as(
-          "normalizedWeight"),
+        normalizeWeight(col("maxWeight"), col("elementCount"), lit(rev.desiredCubeSize))
+          .as("normalizedWeight"),
         col("files"))
       .as[CubeStatus]
       .collect()
       .foreach(row => builder += row.cubeId -> row)
     builder.result()
   }
 
+  def buildReplicatedAndAnnouncedSets(
```

Comment:
Do we still need to distinguish between replicated and announced? isn't better to manage them together?

Reply:
Certainly, it could be some hidden code. As far as I know and remember, the Announced Set was retrieved from the `Keeper` and used for writing new data. The "ANNOUNCED" state on the files was only processed by the `QueryExecutor` in order to filter files.

Since there is only one state for the cube, I think it's best to align the code and not keep ghosts.

Reply:
It seems that we cannot simply merge them. Let's consider the following scenario:

1. IndexStatus is loaded from the log, so some cubes have all their files replicated and some have some their files replicated. The first ones go to the replicated set and the second ones to the announced set
2. The announced set is extended by the cubes reported by the keeper as announced.
3. Since the keeper is considered to be unreliable it can report some already replicated cubes as if they were still announced, so the cubesToOptimize set is computed as a difference between the announced set and replicated set.

If merge announced and replicated sets then there is a risk to replicate some cube twice. So suggest to leave it as is.

Reply:
But I think that we discussed that there were no "Replicated" cubes. Since cubes are either declared Announced or not, we don't have a "fully replicated" set of cubes and avoid this type of conflict. 

The problem is to have two sets of "announced" cubes, the one for the Keeper and the one from the Log. The one retrieved from the Keeper should contain the one from the Log, so yes, it could be duplicated. But even if `cubesToOptimize` contains an "already replicated" cube, it does not matter because we are going to take a look at the files before replicating. Is it correct?

Reply:
Yes, if all the files are already replicated then there is nothing to do during the replication. Yes, it can be checked at the later stage when we try to replicate a cube, however the existing code does it earlier and it works, so I prefer to leave this particular part as is. However if you @cugni , @osopardo1 prefer the approach with late check I will rework it.

Reply:
Oh, I see. For me it's a matter of understandability of what's proposed: to change the number of states of the cubes and the files. 

No hard constraints. If you think is safer to leave the code as it is, we can acknowledge and rework it another time.